unity/EVPreprocessing/Assets/Editor/RecursiveDeleter.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+
+public class RecursiveDeleter 
+{
+    private static readonly log4net.ILog Log = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
+
+    public static void DeleteRecursivelyWithSleep(string destinationPath)
+    {
+        const int tries = 10;
+
+        for (var attempt = 1; attempt <= tries; attempt++)
+        {
+            try
+            {
+                TryDeleting(destinationPath);
+            }
+            catch (DirectoryNotFoundException)
+            {
+                return;
+            }
+            catch (FileNotFoundException)
+            {
+                return;
+            }
+            catch (UnauthorizedAccessException)
+            { // Someone or something hasn't closed a file yet.
+                Log.Debug($"Unauthorized Access at path: {destinationPath}, attempt #{attempt}. Sleeping for 50ms and trying again.");
+                Thread.Sleep(50);
+                if (attempt == tries)

```

Comment:
It is required with the way the code is structured now - without continue "attempt++" is unreachable as it would always get out on return at line 40 with the first iteration of the loop. I can't get return out of the loop as it would to crash at the first successful delete (FileNotFoundException), provided it's not the last attempt.